upup/pkg/fi/cloudup/awsup/status.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsup
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"strings"
+)
+
+// FindClusterStatus discovers the status of the cluster, by looking for the tagged etcd volumes
+func (c *awsCloudImplementation) FindClusterStatus(cluster *kops.Cluster) (*kops.ClusterStatus, error) {
+	etcdStatus, err := findEtcdStatus(c, cluster)
+	if err != nil {
+		return nil, err
+	}
+	status := &kops.ClusterStatus{
+		EtcdClusters: etcdStatus,
+	}
+	glog.V(2).Infof("Cluster status (from cloud): %v", fi.DebugAsJsonString(status))
+	return status, nil
+}
+
+// FindEtcdStatus discovers the status of the cluster, by looking for the tagged etcd volumes
+func (c *MockAWSCloud) FindClusterStatus(cluster *kops.Cluster) (*kops.ClusterStatus, error) {
+	etcdStatus, err := findEtcdStatus(c, cluster)
+	if err != nil {
+		return nil, err
+	}
+	return &kops.ClusterStatus{
+		EtcdClusters: etcdStatus,
+	}, nil
+}
+
+// findEtcdStatus discovers the status of etcd, by looking for the tagged etcd volumes
+func findEtcdStatus(c AWSCloud, cluster *kops.Cluster) ([]kops.EtcdClusterStatus, error) {
+	glog.V(2).Infof("Querying AWS for etcd volumes")
+	statusMap := make(map[string]*kops.EtcdClusterStatus)
+
+	tags := c.Tags()
+
+	request := &ec2.DescribeVolumesInput{}
+	for k, v := range tags {
+		request.Filters = append(request.Filters, NewEC2Filter("tag:"+k, v))
```

Comment:
I assume this does not include the master tag?  Should we add it?

I am wondering if we need a tag that is added to both etcd volumes.  Separate from this PR, but it would help a lot.